LogMonitor/src/LogMonitor/EtwMonitor.cpp

```diff
@@ -558,10 +558,35 @@ EtwMonitor::StartTraceSession(
                     pwsProviderId = NULL;
                 }
 
-                if (status == ERROR_NO_SYSTEM_RESOURCES)
+                switch (status)
                 {
+                case ERROR_INVALID_PARAMETER:
+                    logWriter.TraceWarning(L"The ProviderId id NULL or the TraceHandle is 0.");
+                    break;
+                case ERROR_TIMEOUT:
+                    logWriter.TraceWarning(L"The timeout value expired before the enable callback completed.");
+                    break;
+                case ERROR_INVALID_FUNCTION:
+                    logWriter.TraceWarning(L"You cannot update the level when the provider is not registered.");
+                    break;
+                case ERROR_NO_SYSTEM_RESOURCES:
                     logWriter.TraceWarning(L"Exceeded the number of ETW trace sessions that the provider can enable.");
+                    break;
+                case ERROR_ACCESS_DENIED:
+                    logWriter.TraceWarning(L"Only users with administrative privileges can enable event providers to a cross-process session.");
+                    break;
+                default:
+                    break;

```

Comment:
default should log the raw error code so that we know something happened - e.g. An unknown error occurred - <status>.